associations.rb

```diff
@@ -46,26 +50,38 @@ def generate_migrations
 
 
 class Hotel < ActiveRecord::Base
-  #insert our associations here
+  has_many :rooms
+  has_many :bookings, :through => :rooms
  
   def to_s
     "#{name} with #{rooms.count} rooms"
   end
+
+  def booked_guests
+    [].tap do |r|
+      self.bookings.each { |b| r << b.guest }
+    end
```

Comment:
Oh, good point - I didn't think about function performance while I was adding to the models. 
 The `User.booked_rooms` method will run into the same problem then, right?